sources/packages/backend/apps/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -93,58 +97,120 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
       const uniqueFileNames: string[] = studentAppealRequests.flatMap(
         (studentAppeal) => studentAppeal.files,
       );
-      if (uniqueFileNames.length > 0) {
+      if (uniqueFileNames.length) {
         await this.studentFileService.updateStudentFiles(
           studentId,
-          userId,
+          auditUserId,
           uniqueFileNames,
           FileOriginType.Appeal,
           { entityManager: entityManager },
         );
       }
-      const student = application.student;
-      const ministryNotification: StudentSubmittedChangeRequestNotification = {
-        givenNames: student.user.firstName,
-        lastName: student.user.lastName,
-        email: student.user.email,
-        birthDate: student.birthDate,
-        applicationNumber: application.applicationNumber,
-      };
-      const notificationPromise =
-        this.notificationActionsService.saveStudentSubmittedChangeRequestNotification(
-          ministryNotification,
-          entityManager,
-        );
+      const notificationPromise = this.createStudentAppealNotification(
+        studentId,
+        entityManager,
+        applicationId,
+      );
       const studentAppealPromise = entityManager
         .getRepository(StudentAppeal)
         .save(studentAppeal);
-      const [, createdStudentAppeal] = await Promise.all([
-        notificationPromise,
+      const [createdStudentAppeal] = await Promise.all([
         studentAppealPromise,
+        notificationPromise,
       ]);
       return createdStudentAppeal;
     });
   }
 
   /**
-   * Find any pending appeal for the application if exists.
-   * @param applicationId application id related to the appeal.
-   * @returns exist status
+   * Create a notification for the student appeal.
+   * @param studentId student ID to send the notification.
+   * @param entityManager entity manager to keep DB operations in the same transaction.
+   * @param applicationId application ID related to the appeal, when applicable.
    */
-  async hasExistingAppeal(applicationId: number): Promise<boolean> {
-    const existingAppeal = await this.repo
-      .createQueryBuilder("appeal")
-      .select("1")
-      .innerJoin("appeal.appealRequests", "appealRequests")
-      .innerJoin("appeal.application", "application")
-      .where("application.id = :applicationId", { applicationId })
-      .andWhere("appealRequests.appealStatus = :pending", {
-        pending: StudentAppealStatus.Pending,
-      })
-      .limit(1)
-      .getRawOne();
+  private async createStudentAppealNotification(
+    studentId: number,
+    entityManager: EntityManager,
+    applicationId?: number,
+  ): Promise<void> {
```

Comment:
If the method uses appeal id instead, we can have a select query without any condition selecting the student and the application from the appeal.